src/sentry/eventstore/models.py

```diff
@@ -382,8 +382,17 @@ def get_sorted_grouping_variants(self, force_config=None):
     @staticmethod
     def _hashes_from_sorted_grouping_variants(variants):
         """ Create hashes from variants and filter out None values """
-        hashes = (variant.get_hash() for variant in variants)
-        return [hash_ for hash_ in hashes if hash_ is not None]
+        filtered_hashes = []
+        seen_hashes = set()
+        for variant in variants:
+            hash_ = variant.get_hash()
+            if hash_ is None or hash_ in seen_hashes:
+                continue
+
+            seen_hashes.add(hash_)
+            filtered_hashes.append(hash_)
+
+        return filtered_hashes
```

Comment:
This means that any client that doesn't use `get_hashes()` will have to deduplicate the hashes itself, right? I'm thinking of the grouping script.

Reply:
Yes. I believe we may have to adapt the API a bit but don't know how without computing hashes twice